src/__tests__/index.spec.ts

```diff
@@ -0,0 +1,147 @@
+    expect(XHRInterceptor.setSendCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setResponseCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.enableInterception).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.disableInterception).toHaveBeenCalledTimes(0);
+    expect(logger.enabled).toBe(true);
+
+    startNetworkLogging();
+
+    expect(XHRInterceptor.isInterceptorEnabled).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setOpenCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setRequestHeaderCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setHeaderReceivedCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setSendCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.setResponseCallback).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.enableInterception).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.disableInterception).toHaveBeenCalledTimes(0);
+    expect(logger.enabled).toBe(true);
+  });
+
+  it('should stop the logger', () => {
+    (XHRInterceptor.isInterceptorEnabled as jest.Mock).mockReturnValueOnce(
+      false
+    );
+    expect(logger.enabled).toBe(false);
+
+    startNetworkLogging();
+
+    expect(XHRInterceptor.isInterceptorEnabled).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.enableInterception).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.disableInterception).toHaveBeenCalledTimes(0);
+
+    stopNetworkLogging();
+
+    expect(XHRInterceptor.isInterceptorEnabled).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.enableInterception).toHaveBeenCalledTimes(1);
+    expect(XHRInterceptor.disableInterception).toHaveBeenCalledTimes(1);
+
+    expect(XHRInterceptor.setOpenCallback).toHaveBeenCalledTimes(2);
+    expect(XHRInterceptor.setRequestHeaderCallback).toHaveBeenCalledTimes(2);
+    expect(XHRInterceptor.setHeaderReceivedCallback).toHaveBeenCalledTimes(2);
+    expect(XHRInterceptor.setSendCallback).toHaveBeenCalledTimes(2);
+    expect(XHRInterceptor.setResponseCallback).toHaveBeenCalledTimes(2);
+
+    expect(logger.enabled).toBe(false);
+    expect(logger.paused).toBe(false);
+    // @ts-ignore
+    expect(logger.requests).toEqual([]);
+    // @ts-ignore
+    expect(logger.xhrIdMap).toEqual({});
+    // @ts-ignore
+    expect(logger.maxRequests).toBe(500);
+    // @ts-ignore
+    expect(logger.ignoredHosts).toBeUndefined();
+    // @ts-ignore
+    expect(logger.ignoredUrls).toBeUndefined();
+    // @ts-ignore
+    expect(logger.ignoredPatterns).toBeUndefined();
+  });
+});
+
+describe('clearRequests', () => {
+  it('should clear the requests', () => {
+    logger.callback = jest.fn();
+
+    // @ts-ignore
+    logger.requests = ['test-request'];
+
+    logger.clearRequests();
+
+    expect(logger.getRequests()).toEqual([]);
+    expect(logger.callback).toHaveBeenCalledTimes(1);
+    expect(logger.callback).toHaveBeenCalledWith([]);
+  });
+});
```

Comment:
update after https://github.com/alexbrazier/react-native-network-logger/pull/90 is merged

```suggestion
describe('clearRequests', () => {
  it('should clear the requests', () => {
    jest.useFakeTimers();
    logger.callback = jest.fn();

    // @ts-ignore
    logger.requests = ['test-request'];

    logger.clearRequests();
    jest.advanceTimersByTime(LOGGER_REFRESH_RATE);

    expect(logger.getRequests()).toEqual([]);
    expect(logger.callback).toHaveBeenCalledTimes(1);
    expect(logger.callback).toHaveBeenCalledWith([]);
    jest.useFakeTimers();
  });
});
```